core/include/scipp/core/multi_index.h

```diff
@@ -150,13 +165,37 @@ template <scipp::index N> class SCIPP_CORE_EXPORT MultiIndex {
       increment_outer();
   }
 
+  constexpr void increment_inner_by(const scipp::index distance) noexcept {
+    for (scipp::index data = 0; data < N; ++data) {
+      m_data_index[data] += distance * m_stride[data][0];
+    }
+    m_coord[0] += distance;
+  }
+
+  [[nodiscard]] auto inner_strides() const noexcept {
+    std::array<scipp::index, N> strides;
+    for (scipp::index data = 0; data < N; ++data) {
+      strides[data] = m_stride[data][0];
+    }
+    return strides;
```

Comment:
Would it make sense to transpose `m_strides`, at least the latter method would become simpler? Would also be more efficient for access in `increment()`?

Reply:
Not sure if this is more efficient, the array is pretty small.
But most operations would be more natural if it were transposed. And the other operations are executed less often.